common/src/main/java/org/apache/comet/CometSchemaImporter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.comet;
+
+import org.apache.arrow.c.*;
+import org.apache.arrow.memory.BufferAllocator;
+
+/** This is a simple wrapper around SchemaImporter to make it accessible from Java Arrow. */
+public class CometSchemaImporter extends AbstractCometSchemaImporter {
```

Comment:
This moves `CometSchemaImporter` out of the Arrow namespace (which gets shaded in Iceberg).

Reply:
The original class is now an abstract base class and is still in the arrow namespace so that it can access private members. My hack was then to add the concrete class to extend it in the Comet namespace.

I agree that we should do more to refactor this so that only essential parts are in Arrow namespace. I can file as issue later today.